packages/shared/src/utils/validation/declarative-validation-rules.ts

```diff
@@ -1,37 +1,46 @@
 import { addComma } from '../currency';
 import { cloneObject } from '../object';
 import { compareBigUnsignedInt } from '../string';
+import { TFormErrorMessagesTypes } from './form-error-messages-types';
 
-const validRequired = (value /* , options, field */) => {
+export type TOptions = {
+    min: number;
+    max: number;
+    type?: string;
+    decimals?: string | number;
+    regex?: RegExp;
+};
+
+const validRequired = (value?: string /* , options, field */) => {
     if (value === undefined || value === null) {
         return false;
     }
 
-    const str = String(value).replace(/\s/g, '');
+    const str = value.replace(/\s/g, '');
```

Comment:
```suggestion
    const str = String(value).replace(/\s/g, '');
```
this is what's gonna fix the 'value.replace is not a function' error 😊 The thing is value actually comes here as number too because we validate various values here. So there's not much sense in doing this check outside (in